lib/active_model/serializer/adapter/json_api/link.rb

```diff
@@ -3,30 +3,45 @@ class Serializer
     module Adapter
       class JsonApi
         class Link
-          def initialize(serializer)
+          def initialize(serializer, value)
             @object = serializer.object
             @scope = serializer.scope
+
+            # Use the return value of the block unless it is nil.
+            if value.respond_to?(:call)
+              self._value = instance_eval(&value)
```

Comment:
You're right, there's no reason. There just seemed to be a trend (which I kinda disapprove actually) in AMS to move towards accessors rather than ivars. I'll update it.